packages/circuit-ui/components/ImageInput/ImageInput.tsx

```diff
@@ -88,92 +90,107 @@ const InputWrapper = styled.div`
 const HiddenInput = styled.input(
   ({ theme }) => css`
     ${hideVisually()};
-    &:focus + label {
+    &:focus + label > *:last-child {
       ${focusOutline({ theme })};
     }
   `,
 );
 
 type StyledLabelProps = StyleProps & { isLoading: boolean; invalid: boolean };
 
-const baseLabelStyles = css`
-  border-radius: 12px;
-  overflow: hidden;
-  &:hover {
-    cursor: pointer;
+const baseLabelStyles = ({ theme }: StyleProps) => css`
+  cursor: pointer;
+
+  &::before {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: 0%;
+    width: 100%;
+    height: 100%;
+    border-radius: 12px;
+    pointer-events: none;
+    background-color: ${theme.colors.black};
+    opacity: 0;
+    transition: opacity ${theme.transitions.default};
   }
-`;
 
-const addButtonStyles = ({ theme }: StyledLabelProps) => css`
-  &:hover {
-    & > button {
-      background-color: ${theme.colors.p700};
-      border-color: ${theme.colors.p700};
-    }
+  > *:last-child {
+    transition: box-shadow ${theme.transitions.default};
   }
-  &:active {
-    & > button {
-      background-color: ${theme.colors.p900};
-      border-color: ${theme.colors.p900};
+
+  @supports (-webkit-filter: brightness(1)) or (filter: brightness(1)) {
+    transition: filter ${theme.transitions.default};
+
+    &::before {
+      content: none;
     }
   }
 `;
 
 const invalidLabelStyles = ({ theme, invalid }: StyledLabelProps) =>
   invalid &&
   css`
-    &::after {
-      content: '';
-      position: absolute;
-      top: 0;
-      left: 0%;
-      width: 100%;
-      height: 100%;
-      border-radius: 12px;
-      box-shadow: inset 0 0 0 2px ${theme.colors.danger};
+    > *:last-child {
+      box-shadow: 0 0 0 2px ${theme.colors.danger};
     }
-    &:hover::after {
-      box-shadow: inset 0 0 0 2px ${theme.colors.r700};
+    &:hover > *:last-child {
+      box-shadow: 0 0 0 2px ${theme.colors.r700};
```

Comment:
The reason I had made this `inset` is to continue showing the invalid styles when the input is focused:

`main`:

<img width="737" alt="Screenshot 2021-07-21 at 10 28 14" src="https://user-images.githubusercontent.com/35560568/126459852-a33a0849-bc3a-4129-a42d-10e4c2df9da4.png">

This PR:

<img width="737" alt="Screenshot 2021-07-21 at 10 28 24" src="https://user-images.githubusercontent.com/35560568/126459864-2b2a1921-8a4f-49ba-b0f6-240ce9189768.png">

An extra note: if we compare this state to other similar component, like the `Input`, they're not super consistent because the invalid focused `Input` gets a thicker red outline in place of the `focusOutline`. This could also be an option here moving forward 🙂

Reply:
As we discussed offline, this is a necessary tradeoff: The `img` element doesn't support pseudo-elements to add the outline this way. It needs to be added to the element itself in order to inherit its shape (i.e. border-radius). While it would be possible to add multiple box-shadows, managing the combinations of those would add a significant amount of code complexity that doesn't seem worth it.